### ISSUE TITLE
ruff check --statistics --fix-only shows all errors, not just fixable ones

### ISSUE THREAD

**Issue by paulschreiber** [astral-sh/ruff#27759]
### Summary

`ruff check --statistics --fix-only` shows all errors, not just fixable ones. 

Note also the summary line indicates "no errors fixed" which is unnecessary in the case. Let me know if you want that written up as separate bug.

Example:
```
$ ruff check --statistics --fix-only
32	DTZ011 	call-date-today
 8	RUF059 	unused-unpacked-variable
 7	B006   	mutable-argument-default
 6	PLW1508	invalid-envvar-default
 5	BLE001 	blind-except
 3	N999   	invalid-module-name
...snip...
 1	UP047  	non-pep695-generic-function
No errors fixed (28 fixes available with `--unsafe-fixes`).
```

```
$ ruff check --statistics                      
32	DTZ011 	call-date-today
 8	RUF059 	unused-unpacked-variable
 7	B006   	mutable-argument-default
 6	PLW1508	invalid-envvar-default
 5	BLE001 	blind-except
 3	N999   	invalid-module-name
...
 1	UP047  	non-pep695-generic-function
Found 90 errors.
No fixes available (28 hidden fixes can be enabled with the `--unsafe-fixes` option).
```

### Version

ruff 0.16.2

**Comment by ntBre** [comment]
If I'm interpreting the docs for `--fix-only` correctly:

> Apply fixes to resolve lint violations, but don't report on, or exit non-zero for, leftover violations. Implies `--fix`. Use `--no-fix-only` to disable or `--unsafe-fixes` to include unsafe fixes

I think these flags should probably conflict? With `--fix-only` leftover violations aren't supposed to be reported, which would always leave an empty `--statistics` report. Although maybe the empty report is fine rather than making a conflict error. That's how `--fix` works when all diagnostics are fixed, for example:

```console
$ echo 'import math' > example.py         
$ ruff check --fix example.py --statistics
$
```

But with `--fix-only`, this report should always be empty so a conflict still kind of makes sense to me.

I think the "no errors fixed" line is an intentional hint about `--unsafe-fixes`. That part doesn't seem like a bug to me, unless it appeared when some errors were actually fixed.

**Comment by paulschreiber** [comment]
@ntBre is there an existing/different way to list/count only fixable errors? If not, should that be a separate issue?

**Comment by ntBre** [comment]
I don't think there's a current way to count only the fixable errors, no. `--statistics` kind of does this on its own with the `[*]` indicator for the fixable cases. There was some previous discussion of this long ago in https://github.com/astral-sh/ruff/issues/1353#issuecomment-1365291541, but it doesn't look like another flag was ever added.

Currently you could use the JSON output to get these counts:

```console
$ ruff check --statistics --output-format json --exit-zero |
  jq -r '.[] | select(.fixable_count == .count) | "\(.code)\t\(.count)"'
F401    1
```

As discussed on #1353, I guess I'm curious if this is a common enough use case to support natively in the CLI or if something like the `jq` command above is sufficient.

**Comment by paulschreiber** [comment]
My typical workflow is:
1. fix autofixable errors
1. fix unsafe autofixable errors, one error class at a time
1. fix everything else manually

It would be helpful for (2) if I could see how many were left to fix.

**Comment by ntBre** [comment]
I think if you use `--unsafe-fixes` in combination with `--statistics` you get at least pretty close to what you want:

```console
❯ uvx ruff@latest check --unsafe-fixes --statistics
1208	DTZ001 	[ ] call-datetime-without-tzinfo
 849	C408   	[*] unnecessary-collection-call
 730	RUF012 	[ ] mutable-class-default
 620	BLE001 	[ ] blind-except
 331	SIM117 	[ ] multiple-with-statements
 210	SIM102 	[-] collapsible-if
 166	RUF059 	[*] unused-unpacked-variable
 154	DTZ005 	[ ] call-datetime-now-without-tzinfo
 114	TRY201 	[*] verbose-raise
 110	ISC004 	[*] implicit-string-concatenation-in-collection-literal
```

The `[*]` indicator marks rules where all instances are fixable and `[-]` where only some instances are.

This reminds me of some workflows around an interactive mode (https://github.com/astral-sh/ruff/issues/4361).

**Comment by joshuavetos** [comment]
Would you prefer ⁠--fix-only⁠ and ⁠--statistics⁠ to conflict at the CLI level, or should ⁠--statistics⁠ filter down to only report diagnostics targeted by ⁠--fix-only⁠? Happy to put together a PR for whichever direction makes the most sense.
